libc/src/math/generic/range_reduction_fma.h

```diff
@@ -38,6 +38,16 @@ LIBC_INLINE int64_t small_range_reduction(double x, double &y) {
   return static_cast<int64_t>(kd);
 }
 
+// Return k and y, where
+//   k = round(x * 32) and y = (x * 32) - k.
+//   => pi * x = (k + y) * pi / 32
+LIBC_INLINE int64_t small_range_reduction_mul_pi(double x, double &y) {
```

Comment:
If anyone has a better idea for how to name this function I'd love to hear it

Reply:
This range reduction for `sinpif` is correct even without FMA.  You can move it to `sincosf_utils.h`, or even inlining it in `sincospif_eval`.  Otherwise, it could be named something like`mod_exp2_m32`.